examples/simple-proposals.yaml

```diff
@@ -0,0 +1,262 @@
+    '@type': ProposedIntent
+    publishedIn: fablab:33e8933b-ff73-4a01-964a-ca7a98893083 # the proposal
+    publishes: fablab:60f4204e-b8d2-4026-8577-102c3f82c0af # the intent
+    reciprocal: false
+
+  - '@id': fablab:583e83d9-a46d-44ff-bd71-88513a1d83c0
+    '@type': Intent
+    name: Payment
+    skos:note: Payment when the unit is manufactured and sold
+    action: transfer
+    provider: https://fablab.example/
+    resourceClassifiedAs: https://www.wikidata.org/wiki/Q4917 # US Dollar
+    under: https://fablab.example/soil-sensor-project/value-equation/
+
+  - '@id': fablab:d4d2fd71-34f2-41c3-b1c5-19ad5ed2da59b
+    '@type': ProposedIntent
+    publishedIn: fablab:33e8933b-ff73-4a01-964a-ca7a98893083 # the proposal
+    publishes: fablab:583e83d9-a46d-44ff-bd71-88513a1d83c0 # the intent
+    reciprocal: true
+
+  # Intent as part of 2 proposals
+
+  - '@id': alice:8e5fe80d-a769-4bd5-89e5-2136d33eab9f
+    '@type': Proposal
+    name: Free used bike # alice did not get any responses to her proposal above, so decided to offer the bike for free
+    hasBeginning: 2019-04-01T00:00:00-5:00
+    hasEnd: 2019-05-01T00:00:00-5:00
+    eligibleLocation: map:566a2c01-96f8-4b3c-adcf-f197eced3092 # Madison, WI
+
+  - '@id': alice:a8236bbb-81e0-422d-9861-56d2417db0fb
+    '@type': ProposedIntent
+    publishedIn: alice:8e5fe80d-a769-4bd5-89e5-2136d33eab9f # the proposal
+    publishes: alice:02b39a30-3e04-4305-9656-7f261aa63c84 # the intent (this is the same intent alice originally published)
+
+  # Price list
+
+  - '@id': farm:da7bfa38-a1b2-4f6f-a896-b9bf1ec8807a
+    '@type': Proposal
+    hasBeginning: 2019-03-01T00:00:00-5:00
+    hasEnd: 2019-03-31T00:00:00-5:00
+    eligibleLocation: map:566a2c01-96f8-4b3c-adcf-f197eced3092 # Madison, WI
+
+  - '@id': farm:c7897c39-7f05-4a5d-a487-80e130a24345
+    '@type': Intent
+    name: Carrots case 25#
+    action: transfer
+    provider: https://farm.example/
+    resourceClassifiedAs: wd:Q81 # carrot
+    resourceSpecifiedAs: farm:2402b57c-1c45-423a-acb2-f50961c97d65 # 25 pound case organic carrots seconds
+    unitQuantity:
+      qudt:unit: unit:Number
+      qudt:numericValue: 1
+
+  - '@id': farm:a8236bbb-81e0-422d-9861-56d2417db0fb
+    '@type': ProposedIntent
+    publishedIn: farm:da7bfa38-a1b2-4f6f-a896-b9bf1ec8807a # the proposal
+    publishes: farm:c7897c39-7f05-4a5d-a487-80e130a24345 # the intent
+    reciprocal: false
+
+  - '@id': farm:0f563083-7f05-4a5d-a487-80e130a24345
+    '@type': Intent
+    name: Payment
+    action: transfer
+    receiver: https://farm.example/
+    resourceClassifiedAs: https://www.wikidata.org/wiki/Q4917 # US Dollar
+    unitQuantity:
+      qudt:unit: unit:Number
+      qudt:numericValue: 18
+
+  - '@id': farm:f325a950-5737-488d-b122-8f21114d0eb0
+    '@type': ProposedIntent
+    publishedIn: farm:da7bfa38-a1b2-4f6f-a896-b9bf1ec8807a # the proposal
+    publishes: farm:0f563083-7f05-4a5d-a487-80e130a24345 # the intent
+    reciprocal: true
```

Comment:
on could add here something like

reciprocityQuantity:
     qudt:unit: unit:Number
      qudt:numericValue: 18

Since this stays specific to the proposal but not the intent itself.

Reply:
18 whats?

Reply:
Intent referenced by publishes already deifnes what

Reply:
That is way too confusing and complex - nowhere else do we have quantity separated from quantity of what.  There is no need to calculate an exchange rate.  It is totally defined by the unitQuantity on the intents themselves.

Reply:
Noting on intent can define 'exchange rate' since it stays specific to just one of proposals referencing that intent.

Reply:
Please define 'exchange rate' as you are using it.

Reply:
In this case we have: 18USD for 1 case of carrots

While the very same intent to provide carrots could get used at the same time in proposals like: 1h of work for 2 cases of carrots. And in the same way, the very same intent to receive USD can get used at the same time with proposals like 30USD for 1 case of potatoes. In the example of 'price list' you shared, one should have possibility to have just one intent to receive USD and reuse it in all the proposals for each kind of produce they provide.

Last but not least, 'exchange rate' while sufficient for 'price list' use case oversimplifies things a little. We can have proposals like 500g bar of chocolate for 80MXN + 30LETS (1 providing & 2 receiving), so this non-dual exchange would still scale proportionally each reciprocityQuantity from each ProposedIntent.

Reply:
If I offer my used bike (intent with Lynn provider), and request in return $40 (intent with Lynn receiver), is that an 'exchange rate'?  Still trying to make sure I understand your definition.

Reply:
Yes, 1 (specific) bike for 40USD fits what i would call 'exchange rate' for that specific proposal.

> And in that case, would you put the quantity $40 on Intent or ProposedIntent?

Classification (wd:Q4917 # US Dollar) on Intent and unit:Number & value 40 on ProposedIntent. Agent could use the same intent to receive (wd:Q4917 # US Dollar) with 100 other proposals, and each proposal needs its own specific exchange rate.

Reply:
So then exchange rate applies whenever there is a reciprocal intent of any sort, correct?

Reply:
I wouldn't call intent itself as 'reciprocal' since on itself it just represents some isolated part of agent's plan. Reciprocity appears when we have proposal for exchange, so proposal which references at least one intent where agent publishing proposal wants to *provide* and at least one intent where agent publishing proposal wants to *receive*. Again, intents directly (or flow of any stage in that matter) have no notion of reciprocity, it comes into play first in a proposal and later in agreement made based on that proposal.

Reply:
that's what i meant, of course - intent as part of a proposal - I'd still like to understand if that is how you think about it, I'm trying to think of a compromise way to model this, and hoping to come up with something simpler

> isolated part of agent's plan

that can be a little grandiose, don't you think? it might be what happened to me recently, my mom was asked to move to a smaller memory care unit from her assisted living apartment, and I gave away all her extra furniture when she moved, as quickly as I could so she could stop paying rent on her old apartment

I think it would be good to expand your view of the use cases, to cover what people are actually doing in these transitional times, in addition to ideas for how to put together some new over-arching economic methods...

Reply:
> I think it would be good to expand your view of the use cases

As long as we don't put any proposal specific information directly on intent my approach seems to work just fine with all considered so far use cases.

> I'm trying to think of a compromise way to model this, and hoping to come up with something simpler

Simple check on any alternative approach. Will it work if in your farmer's produce UC agent providing that produce has one and only one intent to receive USD (for example intended USD income for current week long operating cycle).

Reply:
See gitter.

Reply:
>Please also keep in mind that once quantity on intent (I think you call it flowQuantity) gets fully satisfied in commitments, other agents should consider all proposals that reference that intent as finished. In your approach, how do you represent how much of the flowQuantity on intent already got satisfied with commitments?

Since we didn't get to that in our chat, here are some cases where there is no availableQuantity (or flowQuantity) on offers.  All make-to-order scenarios (Dell back in the day, machine job shops, anything custom).  Any producer or distributor where they know they can make or purchase something if they do run out of stock. Any producer where they have known receivers and know they won't run out so it is not relevant to their proposal (like the first price list above). 

Also things change, especially estimates, and that is what the finished flag is for, there will be very few situations where software can make that assumption.  

Generally I think these are some of many reasons to not make assumptions about how intents will work for people, especially until we have way more experience with them.